joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/parser/CdtParser.scala

```diff
@@ -49,12 +49,21 @@ class CdtParser(parseConfig: ParserConfig, headerFileFinder: HeaderFileFinder)
     }
   }
 
+  private def createScannerInfo(file: Path): ScannerInfo = {
+    if (FileDefaults.isCPPFile(file.toString)) {
+      new ScannerInfo(definedSymbols, (includePaths ++ parseConfig.systemIncludePathsCPP).map(_.toString).toArray)
+    } else {
+      new ScannerInfo(definedSymbols, (includePaths ++ parseConfig.systemIncludePathsC).map(_.toString).toArray)
+    }
+  }
```

Comment:
something like this to remove duplication?
```suggestion
    val additionalIncludes = 
      if (FileDefaults.isCPPFile(file.toString)) parseConfig.systemIncludePathsCPP
      else parseConfig.systemIncludePathsC
      
    new ScannerInfo(definedSymbols, (includePaths ++ additionalIncludes).map(_.toString).toArray)
  }
```

Reply:
Done. Thanks!